cfs_config_util/cfs.py

```diff
@@ -376,24 +387,42 @@ def req_payload(self):
         """dict: a dict containing just the layers key used to update in requests"""
         return {'layers': [layer.req_payload for layer in self.layers]}
 
-    def save_to_cfs(self, name=None):
+    def save_to_cfs(self, name=None, overwrite=True):
         """Save the configuration to CFS, optionally with a new name.
 
         Args:
             name (str or None): the name to save as. Required if this
                 configuration does not yet have a name.
+            overwrite (bool): if True, silently overwrite an existing CFS
+                configuration given by `name`. If False, raise a
+                CFSConfigurationError if a configuration with `name` already
+                exists
 
         Returns:
             CFSConfiguration: the new configuration that was saved to CFS
 
         Raises:
             CFSConfigurationError: if there is a failure saving the configuration
-                to CFS.
+                to CFS, or if a CFS configuration exists and `overwrite` is False
         """
         if not name and not self.name:
             raise ValueError('A name must be specified for the CFS configuration.')
         name = name or self.name
 
+        # If overwriting is disabled, throw an error if we try to overwrite a
+        # configuration
+        if not overwrite:
+            try:
+                response = self._cfs_client.get('v2', 'configurations', name, raise_not_ok=False)
+                if response.status_code != 404:
+                    raise CFSConfigurationError(f'A configuration named {name} already exists '
+                                                f'and may not be overwritten.')
+
+            except APIError as err:
+                raise CFSConfigurationError(f'Failed to retrieve CFS configuration "{self.name}": {err}')
+            except ValueError as err:
+                raise CFSConfigurationError(f'Failed to decode JSON response from retrieving '
+                                            f'CFS configuration "{self.name}": {err}')
```

Comment:
I don't believe this is needed as we're not calling `response.json` and don't need to. We just want to check the status code like you're doing.